src/dnne-gen/Generator.cs

```diff
@@ -269,6 +272,10 @@ public void Emit(TextWriter outputStream)
                     }
                 }
 
+                var namespaceString = this.mdReader.GetString(typeDef.Namespace);
+                var classString = this.mdReader.GetString(typeDef.Name);
```

Comment:
Can you confirm this logic works for nested classes? I've added tests that ensure code is generated so it should be easy to quickly validate locally the logic works.

Reply:
I checked this, did some magic  and "now" it works!